### PR TITLE
[IIO] removed blocksize, merging with inner dimension @open sesame 2/4 11:17

### PR DESCRIPTION
**Pull request by kparichay**
1. Removed blocksize (merging multiple buffers of data together). Use tensor_aggregator for this functionality
2. Merging of tensors is done with innermost dimension (merging on outermost dimension introduced reduntant dimensions)
3. Bugfix:
  - buffer size scaled by buffer_capacity
  - data rate_d set to buffer_capacity
4. Moved one-shot mode todo - no support for one-shot mode now

**Related Issue**
#1199 

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

File: gst/nnstreamer/tensor_source/tensor_src_iio.c [hunk 1]
```diff
@@ -443,15 +440,9 @@ gst_tensor_src_merge_tensor_by_type (GstTensorInfo * info, guint size)
...
   }
 
-  /** Now merge into 1 tensor using the outermost dimension*/
-  for (dim_idx = NNS_TENSOR_RANK_LIMIT - 1; dim_idx > 0; dim_idx--) {
-    if (info[base_idx].dimension[dim_idx] != 1) {
-      dim_idx += 1;
-      break;
-    }
-  }
+  /** Now merge into 1 tensor using the innermost dimension*/
```
Comment (myungjoo) [inline gst/nnstreamer/tensor_source/tensor_src_iio.c]: Are you merging n x 4:1:4:1 to 4:n:4:1?

Comment (kparichay) [reply gst/nnstreamer/tensor_source/tensor_src_iio.c]: I am merging n x 4:1:4:1 to 4:n:4:1 (4 x dimension[] = {4, 1, 4, 1} -> dimension[] = {4, 4, 4, 1})
Please correct me if i am wrong - my understanding is dimension[0] is the innermost dimension.

Comment (myungjoo) [reply gst/nnstreamer/tensor_source/tensor_src_iio.c]: Could you please share the rationale why you are merging innermost rank with dimension 1? It appears to be counter-intuitive.

Comment (kparichay) [reply gst/nnstreamer/tensor_source/tensor_src_iio.c]: My primary rationale was - if the innermost dimension remains 1, then it is just adding unnecessary single dimension in the array. Further, merging using the innermost rank makes outer ranks available to future merging/aggregating/formatting of the tensor.

Merging can be done with the outer dimension (n x 1:4:1:1 -> 1:4:n:1) with a squeeze function (1:4:n:1 -> 4:n:1:1). Please let me know if this is more appropriate and I will update?

Comment (myungjoo) [reply gst/nnstreamer/tensor_source/tensor_src_iio.c]: Use decides the dimension. Thus if there is an inner dimension with 1 whose outer dimension is > 1, it is intentional (or we should assume so). Unless there are additional properties dictating which dimension to be merged, I'd suggest to use the innermost outer-dimension of > 1 dimension; e.g.. n x 1:4:1:1 --> 1:4:n:1).

For the flexibility, optionally accepting merging-dimension (rank) looks appropriate as well.

Comment (kparichay) [reply gst/nnstreamer/tensor_source/tensor_src_iio.c]: Updated. Added functionality to merge tensor with innermost/outermost/innermost-outer dimension.

For the IIO data, the data will use dimension `Num of Channels:Buffer Capacity:1:1`.

**Pull request merged**